### PR TITLE
[CELEBORN-434] Add constrain about memory manager's parameters.

### PR DESCRIPTION
**Pull request by FMX**
### What changes were proposed in this pull request?
1. remove range validation.
2. Add constrain to make sure that memory parameters are valid.

### Why are the changes needed?
Ditto.

### Does this PR introduce _any_ user-facing change?
NO.

### How was this patch tested?
UT and cluster run.

### CODE REVIEW

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 1]
```diff
@@ -2761,8 +2760,7 @@ object CelebornConf extends Logging {
       .doc("Max ratio of direct memory to store shuffle data")
       .version("0.2.0")
       .doubleConf
-      .checkValue(v => v >= 0.0 && v <= 0.4, "should be in [0.0, 0.4].")
-      .createWithDefault(0.1)
+      .createWithDefault(0.0)
```
Comment (waitinfuture) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:2763]: Is it OK to default to 0?

Comment (FMX) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:2763]: For version 0.2.0 there is no memory shuffle storage yet. So the default value changed to 0 is safe now.

Comment (waitinfuture) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:2763]: But this PR is for main branch

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks!

**Pull request merged**